sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1237,6 +1239,14 @@ def _decorated(*args, **kwargs):
     self.pvolumes = {}
     self.add_pvolumes(pvolumes)
 
+  @property
+  def is_v2(self):
+    return self._is_v2
+
+  @is_v2.setter
+  def is_v2(self, is_v2: bool):
```

Comment:
This is a public property. In what scenarios are the users expect to set it?

What is the compilation behavior when some ops are v2 and some are not v2?

Perhaps instead of marking individual ops, the compiler should just set global compilation mode while compiling. Then _op_to_template can check the current mode.

WDYT?

Reply:
Ideally, users don't use ContainerOp at all. This is more for the compiler right now.

Reply:
In any case, I do not see a need to make this a public property. It can be a private field.

It's also not really need at ContainerOp at all. What is needed is `compiler/compiler.py` communicating with `compiler/_op_to_template.py`. I'm not sure why they are communicating via `dsl.ContainerOp`.